src/writeData/components/ClientCodeCopyPage/index.tsx

```diff
@@ -25,9 +25,10 @@ const codeRenderer: Renderer<HTMLPreElement> = (props: any): any => {
 
 interface Props {
   contentID: string
+  onCopy?: () => void
 }
 
-const ClientCodeCopyPage: FC<Props> = ({contentID}) => {
+const ClientCodeCopyPage: FC<Props> = ({contentID, onCopy}) => {
```

Comment:
i dont know if we have to prop drill this in here or not. can you ask julia if it's important that this event is fired only by notebooks, or if there's another means of segmenting the data

Reply:
I checked with @juliajames12 and she feels that it's worth it here to make certain this event fires only from Notebooks as it's being used to validate the user flow.